src/panels/VocabularyPanel.tsx

```diff
@@ -266,11 +266,13 @@ export default class ItemPanel extends React.Component<Props, State> {
         this.state.vocabs.find(
           (vocab) => vocab.value === getVocabularyFromScheme(node.scheme)
         ) ||
-        (this.state.vocabs.length === 0 && packageItems.length > 0)
-        || (this.state.vocabs.length === 0 && packageItems.length === 0 && !this.state.search)
+        (this.state.vocabs.length === 0 && vocabularyConcepts.length > 0) ||
+        (this.state.vocabs.length === 0 &&
+          vocabularyConcepts.length === 0 &&
+          !this.state.search)
```

Comment:
When dealing with huge conditions like this, it is often best to assign parts of them to some boolean vars with appropriate name and then use the vars in the if condition, i.e. extract all subterms in parentheses to boolean vars.